sequencer/src/main/java/net/consensys/linea/jsonrpc/JsonRpcManager.java

```diff
@@ -0,0 +1,286 @@
+        return false;
+      }
+
+      // process the response body here ...
+      final String responseBody = response.body() != null ? response.body().string() : null;
+      if (responseBody == null) {
+        log.error("Unexpected empty response body from rejected-tx endpoint");
+        return false;
+      }
+
+      final JsonNode jsonNode = objectMapper.readTree(responseBody);
+      if (jsonNode == null) {
+        log.error("Failed to parse JSON response from rejected-tx endpoint: {}", responseBody);
+        return false;
+      }
+      if (jsonNode.has("error")) {
+        log.error("Error response from rejected-tx endpoint: {}", jsonNode.get("error"));
+        return false;
+      }
+      // Check for result
+      if (jsonNode.has("result")) {
+        final String status = jsonNode.get("result").get("status").asText();
+        log.debug("Rejected-tx JSON-RPC call successful. Status: {}", status);
+        return true;
+      }
+
+      log.warn("Unexpected rejected-tx JSON-RPC response format: {}", responseBody);
+      return false;
+    } catch (final IOException e) {
+      log.error("Failed to send JSON-RPC call to rejected-tx endpoint {}", rejectedTxEndpoint, e);
+      return false;
+    }
+  }
+
+  /**
+   * Saves the given JSON content to a file in the rejected transactions RPC directory. The filename
+   * is generated using a high-precision timestamp and a UUID to ensure uniqueness.
+   *
+   * <p>The file naming format is: rpc_[timestamp]_[uuid].json
+   *
+   * @param jsonContent The JSON string to be written to the file.
+   * @param rejTxRpcDirectory The directory where the file should be saved.
+   * @return The Path object representing the newly created file.
+   * @throws IOException If an I/O error occurs while writing the file, including unexpected file
+   *     collisions.
+   */
+  @VisibleForTesting
+  static Path saveJsonToDir(final String jsonContent, final Path rejTxRpcDirectory)
+      throws IOException {
+    final String timestamp = generateTimestampWithNanos();
+    final String uuid = UUID.randomUUID().toString();
+    final String fileName = String.format("rpc_%s_%s.json", timestamp, uuid);
+    final Path filePath = rejTxRpcDirectory.resolve(fileName);
+
+    try {
+      return Files.writeString(filePath, jsonContent, StandardOpenOption.CREATE_NEW);
+    } catch (final FileAlreadyExistsException e) {
+      // This should never happen with UUID, but just in case
+      log.warn("Unexpected JSON-RPC filename collision occurred: {}", filePath);
+      throw new IOException("Unexpected file name collision", e);
+    }
+  }
+
+  private static String generateTimestampWithNanos() {
+    long millis = System.currentTimeMillis();
+    long nanos = System.nanoTime();
+    long millisPart = millis % 1000;
+    long nanosPart = nanos % 1_000_000;
+    return String.format("%d%03d%06d", millis / 1000, millisPart, nanosPart);
+  }
```

Comment:
curious of this algo, could you explain it?

Reply:
It's basically to generate `[seconds since epoch][milliseconds (000-999)][nanoseconds(000000-999999]`.
Following is the sample millis and nanos values, followed by the final value:
```
Millis: 1726092954875
Nanos: 2315840746018875
1726092954_875_018875
```
We could've avoided the millisecond part calculation and avoid `%03d` as well, since I was calculating the nano part from nano time, I kept the milliseconds part.

Reply:
After some "re-searching" I change the logic to use Instant.now(). Much cleaner!